testsuite/MDAnalysisTests/coordinates/test_gro2.py

```diff
@@ -117,3 +197,82 @@ def __init__(self, reference=None):
         if reference is None:
             reference = GROBZReference()
         super(TestGROBZ2Writer, self).__init__(reference)
+
+
+class GROLargeReference(GROReference):
+    def __init__(self):
+        super(GROLargeReference, self).__init__()
+        self.trajectory = GRO_large
+        self.topology = GRO_large
+
+
+class TestGROLargeWriter(BaseWriterTest):
+    def __init__(self, reference=None):
+        if reference is None:
+            reference = GROLargeReference()
+        super(TestGROLargeWriter, self).__init__(reference)
+
+    @dec.slow
+    @attr('issue')
+    def test_writer_large(self):
+        """
+        Test that atom numbers are truncated for large
+        GRO files (Issue 550).
+        """
+        outfile = self.tmp_file('outfile1.gro')
+        u = mda.Universe(self.ref.topology, self.ref.trajectory)
+        u.atoms.write(outfile)
+
+        with open(outfile, 'rt') as mda_output:
+            with mda.lib.util.anyopen(self.ref.topology, 'rt') as expected_output:
+                produced_lines = mda_output.readlines()[1:]
+                expected_lines = expected_output.readlines()[1:]
+                assert_equal(produced_lines,
+                             expected_lines,
+                             err_msg="Writing GRO file with > 100 000 "
+                                     "coords does not truncate properly.")
+
+    @dec.slow
+    @attr('issue')
+    def test_writer_large_residue_count(self):
+        """
+        Ensure large residue number truncation for
+        GRO files (Issue 886).
+        """
+        outfile = self.tmp_file('outfile2.gro')
+        u = mda.Universe(self.ref.topology, self.ref.trajectory)
+        target_resname = u.residues[-1].resname
+        resid_value = 9999999
+        u.residues[-1].resid = resid_value
+        u.atoms.write(outfile)
+
+        with open(outfile, 'rt') as mda_output:
+            output_lines = mda_output.readlines()
+            produced_resid = output_lines[-2].split(target_resname)[0]
+            expected_resid = str(resid_value)[:5]
+            assert_equal(produced_resid,
+                         expected_resid,
+                         err_msg="Writing GRO file with > 99 999 "
+                                 "resids does not truncate properly.")
+
+
+class TestGROTimestep(BaseTimestepTest):
+    Timestep = mda.coordinates.GRO.Timestep
+    name = "GRO"
+    has_box = True
+    set_box = True
+    unitcell = np.array([10., 11., 12.,
```

Comment:
@kain88-de Most of what was written there (except the equations) didn't make sense to me. 😞